File: solutions/c/resistor-color/1/resistor_color.c
#include "resistor_color.h"

const resistor_band_t COLORS[] = {
    BLACK,
    BROWN,
    RED,
    ORANGE,
    YELLOW,
    GREEN,
    BLUE,
    VIOLET,
    GREY,
    WHITE,
};

const resistor_band_t *colors() {
    return COLORS;
}

File: solutions/typescript/resistor-color/1/resistor-color.ts
const COLORS_MAP = {
  'black': 0,
  'brown': 1,
  'red': 2,
  'orange': 3,
  'yellow': 4,
  'green': 5,
  'blue': 6,
  'violet': 7,
  'grey': 8,
  'white': 9,
};

export const colorCode = (c: keyof typeof COLORS_MAP) => COLORS_MAP[c];
export const COLORS = Object.keys(COLORS_MAP);